### PR TITLE
✨ NEW: Allow for parsing setup.cfg

### PR DESCRIPTION
**Pull request by chrisjsewell**
Ok, so I ended up re-writing a lot of the package lol:

- Restructure the fetching/parsing code, to be more modular
- Add tests for parsing different file types
- Add sourcing of `development_status` from the PyPI classifiers (so it will update dynamically)
  - I also added a warning if the `development_status` key is present and differs from the classifiers
  - and also added an alpha badge, since alpha was assigned to the beta badge for some reason
- Add parsing of setup.cfg
- Allow the use of a `version_file` in `plugin.json, pointing to a file containing the `__version__` variable.
  - This can be used by flit builds and setuptools that use the `version = attr: aiida_plugin.__version__` format.
  - Means you don't have to mess around checking consistency between versions 😄 
- Added `aiida-firecrest` to `plugins.json`, which basically utilises all of this

### CODE REVIEW

File: plugins.json [hunk 1]
```diff
@@ -508,5 +508,12 @@
         "plugin_info": "https://raw.githubusercontent.com/aiidaplugins/aiida-flexpart/main/setup.json",
         "code_home": "https://github.com/aiidaplugins/aiida-flexpart",
         "pip_url": "aiida-flexpart"
+    },
+    "aiida-firecrest": {
+        "entry_point_prefix": "firecrest",
+        "plugin_info": "https://raw.githubusercontent.com/aiidateam/aiida-firecrest/main/setup.cfg",
+        "version_file": "https://raw.githubusercontent.com/aiidateam/aiida-firecrest/main/aiida_firecrest/__init__.py",
+        "code_home": "https://github.com/aiidateam/aiida-firecrest",
+        "pip_url": "aiida-firecrest"
```
Comment (sphuber) [inline plugins.json:517]: Think this shouldn't be in here, right? Probably from another PR

Comment (ltalirz) [reply plugins.json:517]: It's using the setup.cfg, so it's testing the added code.

Fine for me to add it here

**Review by ltalirz — CHANGES_REQUESTED** [review]
Thanks a lot for these additions @chrisjsewell!

The code looks fine to me.

> Add sourcing of development_status from the PyPI classifiers (so it will update dynamically)
I also added a warning if the development_status key is present and differs from the classifiers

Thanks, opened https://github.com/aiidateam/aiida-registry/issues/191 as a follow-up


> and also added an alpha badge, since alpha was assigned to the beta badge for some reason

Yes, we originally thought we don't need the full range of trove classifiers for the registry.
Anyhow, it makes sense to adopt the standard here.

> Allow the use of a version_file in plugin.json, pointing to a file containing the version` variable

Please document this in the `README.md`. After this I'm happy to merge

**Comment by chrisjsewell** [comment]
> Please document this in the README.md. After this I'm happy to merge

Cheers, will do

**Comment by ltalirz** [comment]
> Cheers, will do

I'll take care of it in order to move ahead - you've got enough to do at the moment ;-)

**Pull request merged**